test/test_jit.py

```diff
@@ -254,6 +254,49 @@ def f(x, y):
         self.assertExpectedGraph(trace)
         self.assertExportImport(trace, (x, y))
 
+    def test_peephole(self):
+        a = torch.tensor([0.4], requires_grad=True)
+        b = torch.tensor([0.7], requires_grad=True)
+        c = torch.tensor([0.4, 0.5], requires_grad=True)
+        d = torch.tensor([0], dtype=torch.int32)
+
+        def f(x, y):
+            return x.type_as(y)
+
+        trace, z = torch.jit.get_trace_graph(f, (a, b))
+        self.run_pass('peephole', trace)
+        self.assertExpectedGraph(trace, subname="same_size")
+        trace, z = torch.jit.get_trace_graph(f, (a, c))
+        self.run_pass('peephole', trace)
+        self.assertExpectedGraph(trace, subname="different_size")
+        trace, z = torch.jit.get_trace_graph(f, (a, d))
+        self.run_pass('peephole', trace)
+        self.assertExpectedGraph(trace, subname="different_type")
+
+    def test_peephole_dynamic(self):
+        def f(x, y):
+            return x.type_as(y)
+
+        fn = torch.jit.script(f)
+        torch._C._jit_pass_peephole(fn.graph)
+        self.assertExpectedGraph(fn.graph)
```

Comment:
This is unnecessary. You're testing a pass, so it doesn't matter how you obtained the graph, and there's no need to try both script and trace

Reply:
The reason why I'd like to test that case is to make sure we have the expected behavior when the type_as is trying to cast between dynamic types, which is supposed to be a no-op. By running in script mode,  I could obtain the unannotated graph so that we can test the case as expected.

Reply:
That makes sense. Can we then only use the script? What's the benefit of tracing in this case?

Reply:
Sorry I didn't quite get it. I thought I was only using script here. What do you mean by both script and trace?

Reply:
You're using tracing above. What I mean is that those tests duplicate themselves by only differing in the frontend they use, which shouldn't really matter for this pass.

Reply:
Oh. I think tracing will give me a type annotated graph which prevents me from testing the case I want.

Reply:
You're using tracing above. What I mean is that those tests duplicate themselves by only differing in the frontend they use, which shouldn't really matter for this pass.

Reply:
They matter in that I can use script to get an unannotated graph which I couldn't get in trace mode.

Reply:
I agree. I just don't see much value in ensuring that having no type annotations prevents the optimization (which is pretty obvious, and it's not even the setup in which the pass will be used).

Reply:
I just want to make sure we are not removing useful ops here so that the semantics will remain the same.